packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts

```diff
@@ -280,79 +305,298 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
-				};
-
-				// Create a summarizer client that will be used to summarize the container.
-				const { summarizer: summarizer1 } = await createSummarizer(
-					provider,
-					mainContainer,
-					{
-						runtimeOptions: {
-							gcOptions: { inactiveTimeoutMs },
-						},
-					},
+		describe("Interactive (non-summarizer) clients", () => {
+			/**
+			 * We manufacture a handle to simulate a bug where an object is unrefenced in GC's view
+			 * (and reminder, interactive clients never update their GC data after loading),
+			 * but someone still has a handle to it.
+			 *
+			 * It's possible to achieve this truly with multiple clients where one revives it mid-session
+			 * after it was unreferenced for the inactive timeout, but that's more complex to implement
+			 * in a test and is no better than this approach
+			 */
+			function manufactureHandle<T>(
+				handleContext: IFluidHandleContext,
+				unreferencedId: string,
+			): IFluidHandle<T> {
+				const serializer = new FluidSerializer(handleContext, () => {});
+				const handle: IFluidHandle<T> = parseHandles(
+					{ type: "__fluid_handle__", url: unreferencedId },
+					serializer,
 				);
+				return handle;
+			}
 
-				// Create a data store, mark it as referenced and then unreferenced; summarize;
-				const dataStore = await requestFluidObject<ITestDataObject>(
-					await containerRuntime.createDataStore(TestDataObjectType),
-					"",
-				);
-				const url = dataStore.handle.absolutePath;
-				defaultDataStore._root.set("dataStore", dataStore.handle);
-				defaultDataStore._root.delete("dataStore");
-
-				// Summarize the container. This summary will be used to load another container.
-				const summaryVersion1 = await waitForSummary(summarizer1);
+			const waitForSummary = async (summarizer: ISummarizer) => {
+				await provider.ensureSynchronized();
+				const summaryResult = await summarizeNow(summarizer);
+				return summaryResult.summaryVersion;
+			};
+
+			/**
+			 * Our partners use ContainerRuntime.resolveHandle to issue requests. We can't easily call it directly,
+			 * but the test containers are wired up to route requests to this function.
+			 * (See the innerRequestHandler used in LocalCodeLoader for how it works)
+			 */
+			async function containerRuntime_resolveHandle(
+				container: IContainer,
+				request: IRequest,
+			): Promise<IResponse> {
+				return container.request(request);
+			}
 
-				// Wait for inactive timeout. This will ensure that the unreferenced data store is inactive.
-				await waitForInactiveTimeout();
+			type InactiveLoadError =
+				| (Error & {
+						code: number;
+						underlyingResponseHeaders?: {
+							[InactiveResponseHeaderKey]: boolean;
+						};
+				  })
+				| undefined;
+
+			/**
+			 * 1. With option, handle.get, throws error with known flag
+			 * 2. With option, with header, ok
+			 * 3. Without option, handle.get, ok
+			 *
+			 * Bonus:
+			 * 1. Summarizer never throws
```

Comment:
Is there a test for this - summarizer never throws?

Reply:
Yeah the earlier change in this file that checks the IResponse. I'll delete this comment btw